modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java

```diff
@@ -423,18 +449,263 @@ public Response blacklistApplications(@HeaderParam("Accept") String acceptHeader
+			log.debug("Invoking encrypt operation");
+		}
+
+		MediaType responseMediaType = AndroidAPIUtils.getResponseMediaType(acceptHeader);
+		Message message = new Message();
+
+		try {
+			LockCode lockCode = lockCodeBeanWrapper.getOperation();
+
+			if (lockCode == null) {
+				throw new OperationManagementException("Encrypt bean is empty");
+			}
+
+			ProfileOperation operation = new ProfileOperation();
+			operation.setCode(AndroidConstants.OperationCodes.CHANGE_LOCK_CODE);
+			operation.setType(Operation.Type.PROFILE);
+			operation.setPayLoad(lockCode.toJSON());
+
+			return AndroidAPIUtils.getOperationResponse(lockCodeBeanWrapper.getDeviceIDs(),
+					operation, message, responseMediaType);
+
+		} catch (OperationManagementException e) {
+			message.setResponseMessage("Issue in retrieving operation management service instance");
+			throw new AndroidOperationException(message, responseMediaType);
+		} catch (DeviceManagementException e) {
+			message.setResponseMessage("Issue in retrieving device management service instance");
+			throw new AndroidOperationException(message, responseMediaType);
+		} catch (OperationConfigurationException e) {
+			message.setResponseMessage("Issue in setting up payload in operation");
+			throw  new AndroidOperationException(message, responseMediaType);
+		}
+	}
+
+	@POST
+	@Produces(MediaType.APPLICATION_JSON)
+	@Path("password-policy")
+	public Response setPasswordPolicy(@HeaderParam("Accept") String acceptHeader,
+								   PasswordPolicyBeanWrapper passwordPolicyBeanWrapper) {
+		if (log.isDebugEnabled()) {
+			log.debug("Invoking password policy operation");
+		}
+
+		MediaType responseMediaType = AndroidAPIUtils.getResponseMediaType(acceptHeader);
+		Message message = new Message();
+
+		try {
+			PasswordPolicy passwordPolicy = passwordPolicyBeanWrapper.getOperation();
+
+			if (passwordPolicy == null) {
+				throw new OperationManagementException("Password policy bean is empty");
+			}
+
+			ProfileOperation operation = new ProfileOperation();
+			operation.setCode(AndroidConstants.OperationCodes.PASSWORD_POLICY);
+			operation.setType(Operation.Type.PROFILE);
+			operation.setPayLoad(passwordPolicy.toJSON());
+
+			return AndroidAPIUtils.getOperationResponse(passwordPolicyBeanWrapper.getDeviceIDs(),
+					operation, message, responseMediaType);
+
+		} catch (OperationManagementException e) {
+			message.setResponseMessage("Issue in retrieving operation management service instance");
+			throw new AndroidOperationException(message, responseMediaType);
+		} catch (DeviceManagementException e) {
+			message.setResponseMessage("Issue in retrieving device management service instance");
+			throw new AndroidOperationException(message, responseMediaType);
+		} catch (OperationConfigurationException e) {
+			message.setResponseMessage("Issue in setting up payload in operation");
+			throw  new AndroidOperationException(message, responseMediaType);
+		}
+	}
+
+	@POST
+	@Produces(MediaType.APPLICATION_JSON)
```

Comment:
while the response is built according to the content-type specified by the accept-header, can we just limit the response to JSON by specifying "@Produces(MediaType.APPLICATION_JSON)"?